src/opencast.tsx

```diff
@@ -262,6 +262,24 @@ export class Opencast {
     return await this.jsonRequest("info/me.json");
   }
 
+  /** Returns the value from user.name from the `/info/me.json` endpoint. */
+  async getUsername(): Promise<string | null> {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let request: any;
+    try {
+      request = await this.getInfoMe();
+      if (request) {
+        return request.user.name as string;
+      }
+    } catch (e) {
+      // If it's not our own error, rethrow it.
+      if (!(e instanceof RequestError)) {
+        throw e;
+      }
+    }
+    return null;
```

Comment:
I would use `this.#currentUser` here. That's the result of `info/me.json` the last time the endpoint was requested. This means that we don't have a username if the endpoint was not requested yet at all, e.g. if there are no opencast connection settings set yet. But I think that only happens very rarely and shouldn't be a problem. Not having to wait for a request is worth it, I think.

Also, `this#currentUser` is `unknown`. So for typescript to accept this method, you have to have a bunch of `if (!("user" in this.#currentUser)) { return null; }` checks. If in doubt, feel free to ask me how to do this best.